rust/src/dns/dns.rs

```diff
@@ -790,27 +795,42 @@ pub unsafe extern "C" fn rs_dns_parse_request_tcp(
     flow: *const core::Flow, state: *mut std::os::raw::c_void, _pstate: *mut std::os::raw::c_void,
     stream_slice: StreamSlice, _data: *const std::os::raw::c_void,
 ) -> AppLayerResult {
-    let state = cast_pointer!(state, DNSState);
-    if stream_slice.is_gap() {
-        state.request_gap(stream_slice.gap_size());
-    } else if !stream_slice.is_empty() {
-        return state.parse_request_tcp(flow, stream_slice);
+    match catch_unwind(|| {
+        let state = cast_pointer!(state, DNSState);
+        if stream_slice.is_gap() {
+            state.request_gap(stream_slice.gap_size());
+        } else if !stream_slice.is_empty() {
+            return state.parse_request_tcp(flow, stream_slice);
+        }
+        return AppLayerResult::ok();
+    }) {
+        Ok(result) => result,
+        Result::Err(_) => AppLayerResult::err(),
     }
-    AppLayerResult::ok()
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn rs_dns_parse_response_tcp(
     flow: *const core::Flow, state: *mut std::os::raw::c_void, _pstate: *mut std::os::raw::c_void,
     stream_slice: StreamSlice, _data: *const std::os::raw::c_void,
 ) -> AppLayerResult {
-    let state = cast_pointer!(state, DNSState);
-    if stream_slice.is_gap() {
-        state.response_gap(stream_slice.gap_size());
-    } else if !stream_slice.is_empty() {
-        return state.parse_response_tcp(flow, stream_slice);
+    match catch_unwind(|| {
+	let x: Vec<bool> = vec![];
+        let state = cast_pointer!(state, DNSState);
+        if stream_slice.is_gap() {
+            state.response_gap(stream_slice.gap_size());
+        } else if !stream_slice.is_empty() {
+            return state.parse_response_tcp(flow, stream_slice);
+        }
+	if x[99] {}
```

Comment:
Oops, I didn't mean to commit this introduce error, this is why QA is so far off.